### ISSUE TITLE
QueryDuringMigrationsStressTest.testQueryMapWithIndexes_whileShutdownStartup

### ISSUE THREAD

**Issue by hasancelik** [hazelcast/hazelcast#9804]
```
java.lang.AssertionError: [Obtained 996 results for query 'id >= 53537 AND id < 54537', Obtained 996 results for query 'id >= 24365 AND id < 25365', Obtained 994 results for query 'id >= 21019 AND id < 22019', Obtained 994 results for query 'id >= 93655 AND id < 94655']
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertFalse(Assert.java:64)
	at com.hazelcast.map.QueryDuringMigrationsStressTest$1.run(QueryDuringMigrationsStressTest.java:128)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueAllTheTime(HazelcastTestSupport.java:895)
	at com.hazelcast.map.QueryDuringMigrationsStressTest.queryDuringMigrations(QueryDuringMigrationsStressTest.java:125)
	at com.hazelcast.map.QueryDuringMigrationsStressTest.testQueryMapWithIndexes_whileShutdownStartup(QueryDuringMigrationsStressTest.java:116)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast%20Maintenance/job/Hazelcast-3.maintenance-nightly/com.hazelcast$hazelcast/1123/testReport/junit/com.hazelcast.map/QueryDuringMigrationsStressTest/testQueryMapWithIndexes_whileShutdownStartup/

possibly related to #9724

**Comment by mdogan** [comment]
One of my runs on maintenance branch failed with more than expected entries:

```
testQueryMapWithIndexes_whileShutdownStartup(com.hazelcast.map.QueryDuringMigrationsStressTest)  Time elapsed: 82.807 sec  <<< FAILURE!
java.lang.AssertionError: [Obtained 1006 results for query 'id >= 54255 AND id < 55255', Obtained 1003 results for query 'age >= 55892 AND age < 56892', Obtained 1005 results for query 'age >= 47500 AND age < 48500', Obtained 1003 results for query 'age >= 55724 AND age < 56724', Obtained 1004 results for query 'id >= 79939 AND id < 80939']
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertFalse(Assert.java:64)
	at com.hazelcast.map.QueryDuringMigrationsStressTest$1.run(QueryDuringMigrationsStressTest.java:128)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueAllTheTime(HazelcastTestSupport.java:895)
	at com.hazelcast.map.QueryDuringMigrationsStressTest.queryDuringMigrations(QueryDuringMigrationsStressTest.java:125)
	at com.hazelcast.map.QueryDuringMigrationsStressTest.testQueryMapWithIndexes_whileShutdownStartup(QueryDuringMigrationsStressTest.java:116)
```

**Comment by vbekiaris** [comment]
This is fixed in 3.8. Some changes in MapQueryEngineImpl and the abstract client query task have been backported, but query engine has been refactored heavily so it's hard to locate missing parts of the solution, so closing.
